Eevee.Sleep.Bot/Controllers/Mongo/InGameAnnoucementHistoryControllers.cs

```diff
@@ -0,0 +1,31 @@
+using Eevee.Sleep.Bot.Models;
+using MongoDB.Driver;
+
+namespace Eevee.Sleep.Bot.Controllers.Mongo;
+
+public static class InGameAnnouncememntHistoryController {
+    public static async Task Upsert(InGameAnnouncementDetailModel model) {
```

Comment:
Utilize Mongo transaction, so in case this fails, every other steps in this method can roll back.